uniauth/views.py

```diff
@@ -213,14 +243,17 @@ def cas_login(request, institution):
                 request.session.create()
             auth_login(request, user)
             request.session['auth-method'] = "cas-" + institution.slug
-            return _login_success(request, user, next_url, ["ticket"])
+            return _login_success(request, user, next_url, ["ticket"], cas_auth="cas-" + institution.slug)
 
         # Authentication failed: raise permission denied
         else:
             raise PermissionDenied("Verification of CAS ticket failed.")
 
     # If no ticket was provided, redirect to the
     # login URL for the institution's CAS server
+    elif api_mode: 
+        # return CAS server login URL via Json which will be used to redirect
+        return JsonResponse({'redirect-to': client.get_login_url()})
```

Comment:
💭  We return a lot of different keys depending on the view in question and what happens. We need to make sure these are well documented, so the users of the library know what to expect.